Source/core/Portability.cpp

```diff
@@ -410,5 +395,67 @@ namespace Core {
     {
         toString(dst, format, ap);
     }
-}
-}
+
+    TextFragment Demangled(const char name[]) {
+
+        char allocationName[512];
```

Comment:
I understand we're trying to avoid an extra malloc here, but let's remember this will crash if the demangled name happens to be larger than 511 chars (which is not super hard with C++, but hopefully not common in Thunder) as __cxa_demangle() will attempt a realloc in such case.

Reply:
Every day is a schoolday! I read up on the API specification of the "abi::__cxa_demangle" and indeed it will/can reallocif the buffer is too small. So I guess this is a real issue. GOod to tackle it